File: app/src/main/java/com/example/dcct/presenter/CoverPresenter.java
package com.example.dcct.presenter;

import com.example.dcct.base.BasePresenter;
import com.example.dcct.bean.CoverEntity;
import com.example.dcct.model.CoverModel;
import com.example.dcct.model.Impl.CoverModelImp;
import com.example.dcct.view.ImageUrlCallback;

import java.util.List;

public class CoverPresenter extends BasePresenter {
    private CoverModel mCoverModel = new CoverModelImp();

    public void fetchImageUrl(){
        if (mCoverModel != null && mCallBackView.get() != null) {
            mCoverModel.requestImageUrl( new CoverModel.OnReturnListener() {
                @Override
                public void onComplete(List<CoverEntity> coverEntities) {
                    ((ImageUrlCallback) mCallBackView.get()).onLoadImageUrl( coverEntities );
                }

                @Override
                public void onError(String msg) {
                    ((ImageUrlCallback) mCallBackView.get()).showErrorMsg( msg );
                }
            } );
        }
    }

}

File: app/src/main/java/com/example/dcct/presenter/CoverPresenter.kt
package com.example.dcct.presenter

import com.example.dcct.base.BasePresenter
import com.example.dcct.bean.BackResultData
import com.example.dcct.bean.CoverEntity
import com.example.dcct.model.CoverModel
import com.example.dcct.model.CoverModel.OnReturnListener
import com.example.dcct.model.impl.CoverModelImp
import com.example.dcct.view.ImageUrlCallback

class CoverPresenter : BasePresenter<ImageUrlCallback>() {
    private val mCoverModel: CoverModel = CoverModelImp()
    fun fetchImageUrl() {
        if (mCallBackView!!.get() != null) {
            mCoverModel.requestImageUrl(object : OnReturnListener {
                override fun onComplete(coverEntities: List<CoverEntity>) {
                    mCallBackView!!.get()!!.onLoadImageUrl(coverEntities)
                }

                override fun onError(msg: String) {
                    mCallBackView!!.get()!!.showErrorMsg(msg)
                }
            })
        }
    }
}